bitcoin/src/blockdata/witness.rs

```diff
@@ -298,6 +298,34 @@ impl Witness {
         let pos = decode_cursor(&self.content, self.indices_start, index)?;
         self.element_at(pos)
     }
+
+    /// Get Tapscript following BIP341 rules regarding accounting for an annex.
+    ///
+    /// This does not guarantee that this represents a P2TR [`Witness`].
+    /// It merely gets the second to last or third to last element depending
+    /// on the first byte of the last element being equal to 0x50.
+    pub fn get_tapscript(&self) -> Option<&[u8]> {
+        let len = self.len();
+        self
+            .last()
+            .map(|last_elem| {
+                // From BIP341:
+                // If there are at least two witness elements, and the first byte of
+                // the last element is 0x50, this last element is called annex a
+                // and is removed from the witness stack.
+                if len >= 2 && last_elem.first().filter(|&&v| v == 0x50).is_some() {
```

Comment:
nit: use `const TAPROOT_ANNEX_PREFIX` from taproot.rs

Reply:
stylistic nit(feel free to ignore): Can we directly written as `last_elem.first() == &Some(TAPROOT_ANNEX_PREFIX)`